sklearn/utils/estimator_checks.py

```diff
@@ -3121,6 +3121,66 @@ def check_requires_y_none(name, estimator_orig, strict_mode=True):
             warnings.warn(warning_msg, FutureWarning)
 
 
+def check_n_features_in_after_fitting(name, estimator_orig, strict_mode=True):
```

Comment:
Once all modules are supported, should this be merge with the already-existing `check_n_features_in` check?

We also have another check that specifically checks for error when the number of features are inconsistent (I don't remember the name). Should this one be removed then? (If yes let's document it here and next to N_FEATURES_IN_AFTER_FIT_MODULES_TO_IGNORE please)

Reply:
Yea we should merge it into `check_n_features_in`.

> We also have another check that specifically checks for error when the number of features are inconsistent (I don't remember the name). Should this one be removed then?

I think you are referring to `check_estimators_partial_fit_n_features`. This new check adds two new requirements on top of `check_estimators_partial_fit_n_features`:

1. `n_features_in_` is set during the first call to `partial_fit`.
2. More strict when it comes to the error message.

I updated the comment with the above message.

Reply:
I was more referring to e.g. `check_classifiers_train` :

```py
        msg = ("The classifier {} does not raise an error when the number of "
               "features in {} is different from the number of features in "
               "fit.")
```

but we can keep it as-is and remove later (or not, as long as it passes)